sdcm/ndbench_thread.py

```diff
@@ -0,0 +1,234 @@
+        self.round_robin = round_robin
+        self.params = params if params else dict()
+
+        # remove the ndbench command, and parse the rest of the ; separated values
+        stress_cmd = re.sub(r'^ndbench', '', stress_cmd)
+        self.stress_cmd = ' '.join([f'-Dndbench.config.{param.strip()}' for param in stress_cmd.split(';')])
+        self.stress_cmd = f'./gradlew -Dndbench.config.cli.timeoutMillis={self.timeout * 1000}' \
+                          f' -Dndbench.config.cass.host={self.node_list[0].external_address} {self.stress_cmd} run'
+
+        self.executor = None
+        self.results_futures = []
+        self.max_workers = 0
+        self.shell_marker = generate_random_string(20)
+
+    def run(self):
+        if self.round_robin:
+            self.stress_num = 1
+            loaders = [self.loader_set.get_loader()]
+            LOGGER.debug("Round-Robin through loaders, Selected loader is {} ".format(loaders))
+        else:
+            loaders = self.loader_set.nodes
+
+        self.max_workers = len(loaders) * self.stress_num
+        LOGGER.debug("Starting %d ndbench Worker threads", self.max_workers)
+        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=self.max_workers)
+
+        for loader_idx, loader in enumerate(loaders):
+            for cpu_idx in range(self.stress_num):
+                self.results_futures += [self.executor.submit(self._run_stress, *(loader, loader_idx, cpu_idx))]
+
+        return self
+
+    def create_stress_cmd(self, node, loader_idx):
+        pass
+
+    def _run_stress(self, loader, loader_idx, cpu_idx):
+        log_dir = os.path.join(loader.logdir, self.loader_set.name)
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+        log_file_name = os.path.join(log_dir, f'ndbench-l{loader_idx}-c{cpu_idx}-{uuid.uuid4()}.log')
+        LOGGER.debug('ndbench local log: %s', log_file_name)
+
+        def raise_event_callback(sentinal, line):  # pylint: disable=unused-argument
+            if line:
+                NdbenchStressEvent(type='error', severity=Severity.CRITICAL,
+                                   node=loader, stress_cmd=self.stress_cmd, errors=[str(line)])
+
+        NdbenchStressEvent('start', node=loader, stress_cmd=self.stress_cmd)
+        try:
+
+            LOGGER.debug("running: %s", self.stress_cmd)
+
+            if self.stress_num > 1:
+                node_cmd = f'taskset -c {cpu_idx} bash -c "{self.stress_cmd}"'
+            else:
+                node_cmd = self.stress_cmd
+
+            docker = RemoteDocker(loader, 'scylladb/hydra-loaders:ndbench-jdk8-20200206',
+                                  extra_docker_opts=f'--network=host --label shell_marker={self.shell_marker}')
+
+            node_cmd = f'STRESS_TEST_MARKER={self.shell_marker}; cd /ndbench && {node_cmd}'
+
+            with NdBenchStatsPublisher(loader, loader_idx, ndbench_log_filename=log_file_name):
+                result = docker.run(cmd=node_cmd,
```

Comment:
add try-except like in regular C-S thread

Reply:
added it